client/src/templates/Challenges/classic/editor.tsx

```diff
@@ -1203,8 +1203,16 @@ const Editor = (props: EditorProps): JSX.Element => {
     });
   }
 
-  const { theme } = props;
-  const editorTheme = theme === Themes.Night ? 'vs-dark-custom' : 'vs-custom';
+  const { isSignedIn, theme } = props;
+  const preferDarkScheme = window.matchMedia(
+    '(prefers-color-scheme: dark)'
+  ).matches;
+  const editorTheme =
+    theme === Themes.Night
+      ? 'vs-dark-custom'
+      : preferDarkScheme && !isSignedIn
```

Comment:
```suggestion
    theme === Themes.Night || (preferDarkScheme && !isSignedIn)
```
Nested ternaries are a pain to read and this is more concise.